internal/txprocessor/txprocessor.go

```diff
@@ -69,7 +77,24 @@ func (r TXProcessor) GetSubmitNotificationChannel() <-chan relay.PendingSubmitte
 func (r TXProcessor) submitTxWithProofs(queryID uint64, block *neutrontypes.Block) error {
 	proofStart := time.Now()
 	hash := hex.EncodeToString(tmtypes.Tx(block.Tx.Data).Hash())
-	neutronTxHash, err := r.submitter.SubmitTxProof(queryID, block)
+	var neutronTxHash string
+	err := retry.Do(func() error {
```

Comment:
seems like the current implementation blocks entire query processing workflow. Possible solution is run all processMessage(https://github.com/neutron-org/neutron-query-relayer/blob/feature/resubmit-tx-on-recoverable-error/internal/relay/relayer.go#L79) in the backgtound

Reply:
true, but be attentive to possible race conditions